refactor(test): updated test helpers to be strongly typed

david-driscoll opened the pull request:
**Description:**
This allows tests to pick more on variation in type definitions when tests are being built

**Related issue (if exists):**

benlesh commented:
@david-driscoll it looks like CI is freaking out on this one, but I like the spirit of it... can you have a look?

benlesh commented:
Ping @david-driscoll

kwonoj commented:
Check my proposal here too: https://github.com/ReactiveX/rxjs/pull/3266 - I'd rather moving forward to new test assertion and drop current one once migration's done.

david-driscoll commented:
@kwonoj I've updated and fixed the merging issues, however I'm getting this error in travis:

```
Merge commit detected.
Merge commit detected.
Request failed [404]: https://api.github.com/repos/ReactiveX/rxjs/statuses/369be8144cc597c26ee5808452a2178471fa7ead
Response: {
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/repos/statuses/#create-a-status"
}
Failing the build, there is 1 fail.
```

david-driscoll commented:
I figured it out! 🎉 

This should be good to go now.

benlesh commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@reactivex/rxjs",
-  "version": "6.0.0-alpha.3",
+  "version": "6.0.0-alpha.2",
```
Oops. :D you probably didn't mean to do this.

david-driscoll replied:
yeah I think oj had something merged in, so I just took incoming changes.  Fixed.

benlesh commented on spec/observables/onErrorResumeNext-spec.ts:
```diff
@@ -1,54 +1,50 @@
-    expectSubscriptions(source.subscriptions).toBe(subs);
-  });
-});
+
```
Why is git freaking out about this file? It's going to ruin the blame.

david-driscoll replied:
Appears to be whitespace... the file as it exists today in master is using windows line breaks `\r\n` and when I change the file the `.editorconfig` forces the file to switch to just `\n`.

benlesh commented on spec/observables/throwError-spec.ts:
```diff
@@ -1,11 +1,10 @@
 import { expect } from 'chai';
-import * as Rx from '../../src/Rx';
+import { TestScheduler } from '../../src/internal/testing/TestScheduler';
```
should be `../../src/testing`

david-driscoll replied:
Fixed found 4 instances.

benlesh commented on spec/operators/distinct-spec.ts:
```diff
@@ -1,219 +1,215 @@
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
-  });
+import * as Rx from '../../src/Rx';
```
Another one were git is freaking out... figure out what went wrong if you can.

david-driscoll replied:
CLRF vs LF

benlesh commented on spec/operators/timestamp-spec.ts:
```diff
@@ -1,153 +1,149 @@
-    expectObservable(e1.timestamp(rxTestScheduler)).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-  });
+import * as Rx from '../../src/Rx';
```
Another git freakout. lol

david-driscoll replied:
CLRF vs LF
